dydx/dydxPresenters/dydxPresenters/_v4/MarketInfo/Components/dydxMarketPositionViewPresenter.swift

```diff
@@ -84,5 +77,70 @@ class dydxMarketPositionViewPresenter: HostedViewPresenter<dydxMarketPositionVie
+            viewModel?.takeProfitStatusViewModel = nil
+            viewModel?.stopLossStatusViewModel = nil
+        } else {
+            let decimalDigits = market.configs?.tickSizeDecimals?.intValue ?? 0
+            let stepSizeDecimals = market.configs?.stepSizeDecimals?.intValue ?? 0
+            if takeProfitOrders.count > 1 {
+                viewModel?.takeProfitStatusViewModel = .init(
+                    triggerSide: .takeProfit,
+                    triggerPriceText: DataLocalizer.shared?.localize(path: "APP.TRADE.MULTIPLE_ARROW", params: nil),
+                    action: routeToOrdersAction)
+            } else if let takeProfitOrder = takeProfitOrders.first {
+                viewModel?.takeProfitStatusViewModel = .init(
+                    triggerSide: .takeProfit,
+                    triggerPriceText: dydxFormatter.shared.raw(number: takeProfitOrder.triggerPrice?.doubleValue, digits: decimalDigits),
+                    limitPrice: takeProfitOrder.type == .takeprofitlimit ? dydxFormatter.shared.raw(number: takeProfitOrder.price, digits: decimalDigits) : nil,
+                    amount: position.size?.current?.doubleValue.magnitude == takeProfitOrder.size.magnitude ? nil : dydxFormatter.shared.raw(number: takeProfitOrder.size, digits: stepSizeDecimals),
+                    action: routeToTakeProfitStopLossAction)
+            } else {
+                viewModel?.takeProfitStatusViewModel = .init(
+                    triggerSide: .takeProfit,
+                    action: routeToTakeProfitStopLossAction)
+            }
+
+            if stopLossOrders.count > 1 {
+                viewModel?.stopLossStatusViewModel = .init(
+                    triggerSide: .stopLoss,
+                    triggerPriceText: DataLocalizer.shared?.localize(path: "APP.TRADE.MULTIPLE_ARROW", params: nil),
+                    action: routeToOrdersAction)
+            } else if let stopLossOrder = stopLossOrders.first {
+                viewModel?.stopLossStatusViewModel = .init(
+                    triggerSide: .stopLoss,
+                    triggerPriceText: dydxFormatter.shared.raw(number: stopLossOrder.triggerPrice?.doubleValue, digits: decimalDigits),
+                    limitPrice: stopLossOrder.type == .stoplimit ? dydxFormatter.shared.raw(number: stopLossOrder.price, digits: decimalDigits) : nil,
+                    // don't show amount unless order size is custom
+                    amount: position.size?.current?.doubleValue.magnitude == stopLossOrder.size.magnitude ? nil : dydxFormatter.shared.raw(number: stopLossOrder.size, digits: stepSizeDecimals),
+                    action: routeToTakeProfitStopLossAction)
+            } else {
+                viewModel?.stopLossStatusViewModel = .init(
+                    triggerSide: .stopLoss,
+                    action: routeToTakeProfitStopLossAction)
+            }
+        }
+
+        viewModel?.takeProfitStopLossAction = routeToTakeProfitStopLossAction
+        #endif
```

Comment:
this is all the logic for different button combinations/configurations

lmk if you think i should move to separate VM